docs/examples/ensemble-window.ipynb

```diff
@@ -0,0 +1,429 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "WHSd310cTJGp"
+   },
+   "source": [
+    "# Changepoint Detection Ensembles: Window search method\n",
+    "\n",
+    "<!-- {{ add_binder_block(page) }} -->\n",
+    "\n",
+    "## Introduction\n",
+    "\n",
+    "Many changepoint detection (CPD) procedure can be implemented with `ruptures`. In most case, these procedures involve a search method relying on only one cost function. We are investigating here a way of building an ensemble model, on top of `ruptures` package, using the [window search method](../user-guide/detection/window.md) that will rely on several cost functions. The incentive is that this method takes the best of the methods that consider only one cost. \n",
+    "\n",
+    "The interested reader can refer to [Katser2021](#Katser2021) for a more complete introduction. As they do in the paper, we will make the assumption that the number of changepoints to detect is known in advance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "SfsRv7R7TTpW"
+   },
+   "source": [
+    "## Setup\n",
+    "\n",
+    "First, we make the necessary imports and generate a toy signal. In order to highlight the strength of the ensemble model, we concatenate two kinds of signals. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 441
+    },
+    "id": "xrQQA7izTVnG",
+    "outputId": "42071353-5469-4665-dda0-9e53f99ec5c4"
+   },
+   "outputs": [],
+   "source": [
+    "import time  # for execution time comparison\n",
+    "\n",
+    "import matplotlib.pyplot as plt  # for display purposes\n",
+    "import numpy as np\n",
+    "\n",
+    "import ruptures as rpt  # our package\n",
+    "from ruptures.metrics import randindex\n",
+    "\n",
+    "# generate a signal\n",
+    "n_samples, dim, sigma = 250, 3, 4\n",
```

Comment:
Here, you might have a better showcase if you took the regular `sigma = 1`. 
Indeed, what we see from the results, is that all the costs (included `CostLinear` to some extent) perform pretty poorly on the second half of the signal. 
To show the power of the combination, we might want to have a "better" score on the second half with `CostLinear`. 
WDYT ?

Reply:
@oboulant The problem I see by setting sigma = 1 is that the non-ensemble methods perform too well, which makes the ensemble method useless. This is what I have for `sigma = 1`:
```
ar 0.923
mahalanobis 0.962
l1 0.959
l2 0.962
linear 0.74
ensemble 0.959
```

Reply:
# My puzzle
All right, here are few elements after I had a look into it. I was puzzled that the `linear` cost model was returning really bad scores (in my opinion). Indeed, in my opinion, since the signal is supposed to be tailor made for this cost model, I expected a cleaner score profile on the second half of the signal. See following graph where I highlighted the section I am refering to ⬇️ 
![score_puzzle](https://user-images.githubusercontent.com/12909374/157315224-d296eaac-c4b3-4444-8b10-c4f1afcedf48.png)

# The root problem
So I managed to understand why we witness this behaviour. It is because you normalize on the **complete** signal. If you rather normalize each signal before concatenating them, the score profile for the `linear` cost model is way more clean and how I expect it to be. See following graph ⬇️ 
![bla](https://user-images.githubusercontent.com/12909374/157321551-e6a653c8-0b8e-4f2a-bd06-35bb1f914ef7.jpg)

# Still an issue
The problem is not fully solved since the aggregated score is still not satisfactory according to me. See following graph ⬇️ 
![bli](https://user-images.githubusercontent.com/12909374/157321722-bbcd1c1a-8729-472b-ba94-f9f5a65c2d20.jpg)
I think the aggregated score is not fine because the scaling and aggregating functions do not do their job. But I have to give it some more thoughts and dive deeper into it to identify what is the real issue. 

Please let me know if what I present here is unclear !

Reply:
@oboulant is is really clear. I agree, to my mind, the scaling and aggregating functions are no the best here but they are the ones on which the authors got the best scores:
![Screenshot from 2022-03-09 11-43-03](https://user-images.githubusercontent.com/49516344/157426077-cf50048b-ac9e-460f-8fe6-80614a3581ec.png)

Reply:
The thing is that I think 

1. those figures are really data dependent
2. individual score profiles from the paper seem really different than what we have. Whereas I quite agree on the need for the two steps (scale and aggregate), I do not think than given our individual score profiles,  `MinAbs` is doing the job of scaling. The question of the choice of the aggregating function is more open, since according to me the main failure here is the scaling function

So moving forward I would look into a better way to scale the individual scores. I tried so normalize by the max value because given our individual score at least it allows for comparison afterward. But it fails at the aggregation step because individual scores are noisy (which seems not to be that true if I look at the paper's sample graph) : even with "reasonable" lambas, noise cancels out the signal (on the linear part of the signal). Maybe we have to denoise individual scores before aggregation / before scaling.